client/app/pages/destinations/DestinationsList.jsx

```diff
@@ -0,0 +1,134 @@
+      Destination.types().$promise,
+    ]).then(values => this.setState({
+      destinations: values[0],
+      destinationTypes: values[1],
+      loading: false,
+    }, () => { // all resources are loaded in state
+      if ($route.current.locals.isNewDestinationPage) {
+        this.showCreateSourceDialog();
+      }
+    }));
+  }
+
+  createDestination = (selectedType, values) => {
+    const target = { options: {}, type: selectedType };
+    helper.updateTargetWithValues(target, values);
+
+    return Destination.save(target).$promise.then(() => {
+      this.setState({ loading: true });
+      Destination.query(destinations => this.setState({ destinations, loading: false }));
+    }).catch((error) => {
+      if (!(error instanceof Error)) {
+        error = new Error(get(error, 'data.message', 'Failed saving.'));
+      }
+      return Promise.reject(error);
+    });
+  };
+
+  showCreateSourceDialog = () => {
+    CreateSourceDialog.showModal({
+      types: this.state.destinationTypes,
+      sourceType: 'Alert Destination',
+      imageFolder: IMG_ROOT,
+      onCreate: this.createDestination,
+    }).result.then((success) => {
+      if (success) {
+        this.setState({ loading: true });
+        Destination.query(destinations => this.setState({ destinations }));
+      }
+    }).finally(() => {
+      if ($route.current.locals.isNewDestinationPage) {
+        navigateTo('destinations');
+      }
+    });
+  };
+
+  renderDestinations() {
+    const { destinations } = this.state;
+    const items = destinations.map(destination => ({
+      title: destination.name,
+      imgSrc: `${IMG_ROOT}/${destination.type}.png`,
+      href: `destinations/${destination.id}`,
+    }));
+
+    return isEmpty(destinations) ? (
+      <div className="text-center">
+        There are no alert destinations yet.
+        <div className="m-t-5">
+          <a className="clickable" onClick={this.showCreateSourceDialog}>Click here</a> to add one.
+        </div>
+      </div>
+    ) : (<CardsList items={items} />);
+  }
+
+  render() {
+    return (
+      <div>
+        <div className="m-b-15">
+          <Button type="primary" onClick={this.showCreateSourceDialog}>
+            <i className="fa fa-plus m-r-5" />
+            New Alert Destination
+          </Button>
+        </div>
+        {this.state.loading ? <LoadingState className="" /> : this.renderDestinations()}
+      </div>
+    );
+  }
+}
+
+export default function init(ngModule) {
+  settingsMenu.add({
+    permission: 'admin',
+    title: 'Alert Destinations',
+    path: 'destinations',
+    order: 4,
+  });
+
+  ngModule.component('pageDestinationsList', react2angular(DestinationsList));
+
+  return routesToAngularRoutes([
+    {
+      path: '/destinations',
+      title: 'Alert Destinations',
+      key: 'destinations',
+    },
+    {
+      path: '/destinations/new',
+      title: 'Alert Destinations',
+      key: 'destinations',
+      isNewDestinationPage: true,
```

Comment:
Add `policy.canCreateAlert()` where relevant.

Reply:
`Alert !== AlertDestination` :stuck_out_tongue_closed_eyes: . Though I can add a new policy function for Alert Destinations if you think this is relevant

Reply:
My thinking was that if you can't create an alert you probably can't create a destination as well.
It's only for future-proofing anyway. Whatever you choose.